Ex 02

fabiorodp opened the pull request:
Exercise done!!

Commit:
File file_stats.py created

Commit:
File file_stats.py created

Commit:
Task A - Completed

Commit:
File message_entropy.py created

Commit:
File bubble_sort.py created.

Commit:
Task A - updated and done!

Commit:
Task B - message_entropy.py - done!

Commit:
Task B - updated and Done!

Commit:
Task C - bubble_sort.py - Done!

Commit:
Task C - Bubble_sort - Updated

huynhngoc commented on src/fabio_rodrigues_ex/ex02/file_stats.py:
```diff
@@ -0,0 +1,30 @@
+    2nd. Reads the entire file content into a single string.
+    3rd. Counts how often each character code (0–255) occurs in the string.
+    :return:
+    The result as a list or tuple, where result[i] gives the number of occurrences of character code i.
```
Line's too long here, over 79 characters. This PEP8 violation (I noticed that by email twice).

fabiorodp replied:
> Nice code. Unfortunately PEP8 is violated, which causes an automatic 'fail'. Please remember the code line next time before creating the pull request.

Hi huynhngoc! 

I think this is not fair to fail me because of characters in a comment without give me a second chance to fix it... 

I have done all the procedure from the youtube video made by the professor, included the 79 character violation. I have checked the sign in the top right of my pychamp and everyting was OK... I do not know what happened...

But is it realy unfair to fail me because of that without give me a chance to fix.. This is a really small fix that does not deserve a fail... And I HAVE DONE THE PROCEDURE FROM THE YOUTUBE VIDEO... Something went wrong and I will revise it...

What should I do now? Can I merge the files?

huynhngoc commented on src/fabio_rodrigues_ex/ex02/file_stats.py:
```diff
@@ -0,0 +1,30 @@
+    :return:
+    The result as a list or tuple, where result[i] gives the number of occurrences of character code i.
+    """
+    f = open(textfilename, "r")
```
Would be better if you use `with` statement when opening file.

huynhngoc commented on src/fabio_rodrigues_ex/ex02/message_entropy.py:
```diff
@@ -0,0 +1,28 @@
+import math as m
```
You can import `log2` directly here.

huynhngoc reviewed (commented):
Nice code. Unfortunately PEP8 is violated, which causes an automatic 'fail'. Please remember the code line next time before creating the pull request.

huynhngoc reviewed (approved):
Approved as PEP8 79 characters limit in codeline cannot be detected in Pycharm

Commit:
Updated to use with statement when opening the file

Commit:
Updated to import only log2 fom math

Commit:
It was not a pep8 violation, but I decided to reduce the # of characters anyways

Commit:
It wasnt a pep8 violation, but I decided to reduce the # of characters

Commit:
Merge remote-tracking branch 'origin/ex_02' into ex_02
